src/cosalib/aliyun.py

```diff
@@ -12,37 +12,40 @@
 def remove_aliyun_image(aliyun_id, region):
     print(f"aliyun: removing image {aliyun_id} in {region}")
     try:
-        run_verbose(['ore', 'aliyun', '--log-level', 'debug', 'delete-image',
-                     '--id', aliyun_id,
-                     '--force'])
+        run_verbose([
+            'ore',
+            'aliyun', '--log-level', 'debug', 'delete-image',
+            '--id', aliyun_id,
+            '--force'])
     except SystemExit:
         raise Exception("Failed to remove image")
 
 
-@retry(stop=stop_after_attempt(3))
+@retry(reraise=True, stop=stop_after_attempt(3))
 def aliyun_run_ore_replicate(build, args):
     build.refresh_meta()
     aliyun_img_data = build.meta.get('aliyun', [])
     if len(aliyun_img_data) < 1:
         raise SystemExit(("buildmeta doesn't contain source images. "
                          "Run buildextend-aliyun first"))
 
-    if not args.regions:
-        args.regions = subprocess.check_output([
-            'ore', 'aliyun', 'list-regions'
+    if not args.region:
+        args.region = subprocess.check_output([
+            'ore', f'--config-file={args.config}' if args.config else '',
```

Comment:
Might be nice to create a helper function for the config-file string since it's used in multiple places.

Reply:
Agreed. That's next.